src/python/pants/backend/jvm/tasks/ivy_resolve.py

```diff
@@ -89,15 +90,19 @@ def execute(self):
     if JvmResolveSubsystem.global_instance().get_options().resolver != 'ivy':
       return
 
-    executor = self.create_java_executor()
+    results = []
     targets = self.context.targets()
-    compile_classpath = self.context.products.get_data('compile_classpath',
-        init_func=ClasspathProducts.init_func(self.get_options().pants_workdir))
-    results = self.resolve(executor=executor,
-                                      targets=targets,
-                                      classpath_products=compile_classpath,
-                                      confs=self.get_options().confs,
-                                      extra_args=self._args)
+    # Don't bother creating an executor if there's nothing to do.
+    # This is a bit of an abstraction leak, but executor creation has been observed to take 150ms.
+    if any(target for target in targets if isinstance(target, JarLibrary)):
```

Comment:
this could  be`any(isinstance(target, JarLibrary) for target in targets)`?

Reply:
I think I'd prefer if this were converted into a guard clause, and we skipped the whole ivy task. ie 

```python
if all(target for target in targets if not isinstance(target, JarLibrary)):
  return
```

After the new indent, there's code that depends on the result of the resolve, which could cause issues.

Reply:
Done